packages/tldraw/src/lib/ui/components/menus/TldrawUiMenuCheckboxItem.tsx

```diff
@@ -0,0 +1,98 @@
+import * as _ContextMenu from '@radix-ui/react-context-menu'
+import * as _DropdownMenu from '@radix-ui/react-dropdown-menu'
+import { preventDefault } from '@tldraw/editor'
+import { unwrapLabel } from '../../hooks/useActions'
+import { TLUiEventSource } from '../../hooks/useEventsProvider'
+import { useReadonly } from '../../hooks/useReadonly'
+import { TLUiTranslationKey } from '../../hooks/useTranslation/TLUiTranslationKey'
+import { useTranslation } from '../../hooks/useTranslation/useTranslation'
+import { Icon } from '../primitives/Icon'
+import { Kbd } from '../primitives/Kbd'
+import { useTldrawUiMenuContext } from './TldrawUiMenuContext'
+
+/** @public */
+export function TldrawUiMenuCheckboxItem<
+	TransationKey extends string = string,
+	IconType extends string = string,
+>({
+	id,
+	kbd,
+	label,
+	readonlyOk,
+	onSelect,
+	disabled = false,
+	checked = false,
+}: {
+	icon?: IconType
+	id: string
+	kbd?: string
+	title?: string
+	label?: TransationKey | { [key: string]: TransationKey }
+	readonlyOk: boolean
```

Comment:
trying to parse `if (isReadOnly && !readonlyOk)` hurts my brain :P
can you rename this into something like `isReadonlySomethingSomething` or `shouldReadonly...`?

Reply:
Hm, rather than renaming the variable, I'll make `isReadOnly` -> `isReadOnlyMode`, to differentiate between the editor's mode and the property of the action item.

Reply:
ah i see: my suggestion to say `isReadonlySomethingSomething`/`shouldReadonly` was actually off-base b/c i think i was confused by the naming heh, i was misreading the intent — i thought it was trying to convey that it should "go into read-only mode" vs. "being allowed to work in readonly mode". to that end, i think what would be better is to say `readonlyAllowed` or `allowedInReadonly` to be clearer.

i think it's maybe that "ok" implies a 'soft yes' and that's confusing when combined next to the word 'only' which implies strictness. like, for example, you wouldn't have a variable saying `downloadOnlyOk` because when it's false, that I guess would mean `uploadNeverMaybe`? 🙃 

(this is me getting pedantic on you, i know 😆 )